File: app/src/main/java/org/test/kotlin/demo/repos/ReposFragment.java
package org.test.kotlin.demo.repos;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.test.kotlin.demo.R;
import org.test.kotlin.demo.api.dto.RepositoryDTO;
import org.test.utils.LoadingAdapter;

import java.util.List;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;
import dagger.android.support.DaggerFragment;

public class ReposFragment extends DaggerFragment implements ReposContract.View {

    @Inject
    ReposContract.Presenter presenter;

    public static ReposFragment create(String user) {
        Bundle args = new Bundle(1);
        args.putString(ReposContract.ARG_USERNAME, user);

        ReposFragment fragment = new ReposFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_recycler, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        RecyclerView recyclerView = view.findViewById(R.id.recycler);
        recyclerView.setAdapter(new LoadingAdapter());
    }

    @Override
    public void onStart() {
        super.onStart();

        presenter.onViewReady();
    }

    @Override
    public void showRepositories(@NonNull List<RepositoryDTO> repositories) {
        View view = getView();

        RecyclerView recyclerView = view.findViewById(R.id.recycler);
        recyclerView.setAdapter(new ReposAdapter(repositories));
    }

}

File: app/src/main/java/org/test/kotlin/demo/repos/ReposFragment.kt
package org.test.kotlin.demo.repos

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import dagger.android.support.DaggerFragment
import kotlinx.android.synthetic.main.fragment_recycler.*
import org.test.kotlin.demo.R
import org.test.kotlin.demo.api.dto.RepositoryDTO
import org.test.utils.LoadingAdapter
import org.test.utils.inflate
import javax.inject.Inject

class ReposFragment : DaggerFragment(), ReposContract.View {

    @Inject
    internal lateinit var presenter: ReposContract.Presenter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) =
            container!!.inflate(R.layout.fragment_recycler)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        recycler.adapter = LoadingAdapter()
    }

    override fun onStart() {
        super.onStart()

        presenter.onViewReady()
    }

    override fun showRepositories(repositories: List<RepositoryDTO>) {
        recycler.adapter = ReposAdapter(repositories)
    }

    companion object {

        fun create(user: String) = ReposFragment().apply {
            arguments = bundleOf(ReposContract.ARG_USERNAME to user)
        }

    }

}
